### PR TITLE
Fix safe mode trigger distance

### LINKED ISSUE #18650: Safe Mode Manager max view distance trigger is not max view distance
I've been grappling with the Safe Mode Manager to try to understand it and for a while I thought that maybe it just didn't work -- turns out it's bugged when it comes to specifying trigger distance.

To test the distance where it stops triggering, I spawned a chicken and added it to my blacklist. Here are the rules:

Safe Mode Manager rules

Expected result: Whenever a hostile monster or a chicken are within view, trigger safemode.
Actual result: Whenever a hostile monster or chicken are within 50 tiles, trigger safemode.

When I spawned the chicken, I activated safemode and let it run away from me to see when the safemode alert would trigger. When the chicken was 51 tiles away (but still within view!), safemode stopped triggering.

**Summary: According to the Safe Mode Manager UI, '0' for distance is supposed to be max viewdistance, but in reality only works up to 50 tiles.**

_Additionally, even with the 'test' returning the proper monsters, the * wildcard doesn't seem to trigger on hostile monsters at all._

### PR DESCRIPTION
**Pull request by OzoneH3**
Replaced magic number with constant

Closes #18650

### CODE REVIEW

File: src/safemode_ui.cpp [hunk 1]
```diff
@@ -599,7 +599,7 @@ rule_state safemode::check_monster( const std::string &creature_name_in,
     if( iter != safemode_rules.end() ) {
         const auto &tmp = ( iter->second )[( int )attitude_in];
         if( tmp.state == RULE_BLACKLISTED ) {
-            const int check_dist = ( tmp.proximity == 0 ) ? 50 : tmp.proximity;
+            const int check_dist = ( tmp.proximity == 0 ) ? 60 : tmp.proximity;
```
Comment (codemime) [inline src/safemode_ui.cpp]: I'm pretty sure it's supposed to be a calculable value (or, at the very least, a named constant). If it ends up with a magic number, it'll break again some (probably not too distant) day.

Comment (OzoneH3) [inline src/safemode_ui.cpp]: This check is not even required but I need a value for the SAFEMODEPROXIMITY option which is at max 50 as well atm.
Can't seem to find out how this is actually calculated though.

Comment (OzoneH3) [inline src/safemode_ui.cpp]: Added a constant ```#define MAX_VIEW_DISTANCE ( SEEX * int( MAPSIZE / 2 ) )```

File: src/game_constants.h [hunk 2]
```diff
@@ -36,6 +36,8 @@
 #define SEEX 12
 #define SEEY SEEX
 
+#define MAX_VIEW_DISTANCE ( SEEX * int( MAPSIZE / 2 ) )
```
Comment (OzoneH3) [inline src/game_constants.h]: That would be what, 78?

Pretty sure you can only see 60 tiles. Maybe it should be called differently then? Max render distance!?

**Pull request merged**